DDD.Functions/EloVotingGetPair.cs

```diff
@@ -76,7 +76,7 @@ EloVotingConfig eloVoting
                     Title = s.Title,
                     Abstract = s.Abstract,
                 })
-                .ElementAt(random.Next(receivedSubmissions.Count));
+                .ElementAt(random.Next(receivedSubmissions.Count - 1));
```

Comment:
Actually, I think this might have been fine how it was, or at least the `.Count` wasn't the problem.

[Random.Next(maxValue) does not include the maxValue](https://docs.microsoft.com/en-us/dotnet/api/system.random.next?view=net-6.0#system-random-next(system-int32)) 

Perhaps the problem is because we filter `.Where(x => x.Session != null)` so the count is not actually `receivedSubmissions.Count`

Reply:
Oh even better call, so we'll need to do something else!

Reply:
Ah ok. The `- 1` is OK for `submissionB` (with the assumption that only one session in the list has the same ID as submissionA), but not required for `submissionA`.

```cs
var validSubmissions = receivedSubmissions.Where(x => x.Session != null);
var submissionA = validSubmissions.Select(...).ElementAt(random.Next(validSubmissions.Count);

var validForB = validSubmissions.Where(x => x.Id.ToString() != submissionA.Id);
var submissionB = validForB.Select(...).ElementAt(random.Next(validForB.Count);
```